src/ai/backend/common/docker.py

```diff
@@ -398,83 +341,163 @@ def __eq__(self, other):
-            raise InvalidImageName(self._value)
-        parts = self._value.split("/")
-        if len(parts) == 1:
-            self._registry = default_registry
-            self._name, self._tag = ImageRef._parse_image_tag(self._value, True)
-            if not rx_slug.search(self._tag):
-                raise InvalidImageTag(self._tag, self._value)
-        else:
-            # add ['*'] as magic keyword to accept any repository as valid repo
-            if known_registries == ["*"]:
-                self._registry = parts[0]
-                self._name, self._tag = ImageRef._parse_image_tag("/".join(parts[1:]), False)
-            elif is_known_registry(parts[0], parts[1], known_registries):
-                self._registry = parts[0]
-                using_default = parts[0].endswith(".docker.io") or parts[0] == "docker.io"
-                self._name, self._tag = ImageRef._parse_image_tag(
-                    "/".join(parts[1:]), using_default
-                )
-            else:
-                self._registry = default_registry
-                self._name, self._tag = ImageRef._parse_image_tag(self._value, True)
-            if not rx_slug.search(self._tag):
-                raise InvalidImageTag(self._tag, self._value)
-        self._update_tag_set()
+        image_name = parsed.project_and_image_name.split(f"{project}/")[-1]
 
-    @staticmethod
-    def _parse_image_tag(s: str, using_default_registry: bool = False) -> tuple[str, str]:
-        image_tag = s.rsplit(":", maxsplit=1)
+        return cls(
+            name=image_name,
+            project=project,
+            registry=registry,
+            tag=parsed.tag,
+            architecture=architecture,
+            is_local=is_local,
+        )
+
+    @classmethod
+    def parse_image_tag(
+        cls, image_str: str, *, using_default_repository: bool = False
+    ) -> tuple[str, str]:
+        """
+        Parses the image name and tag from the given image string.
+
+        When the image does not contain '/', and `using_default_repository` is True, it includes the default_repository (lablup) in the image.
+        """
+        image_tag = image_str.rsplit(":", maxsplit=1)
         if len(image_tag) == 1:
-            image = image_tag[0]
+            image_str = image_tag[0]
             tag = "latest"
         else:
-            image = image_tag[0]
+            image_str = image_tag[0]
             tag = image_tag[1]
-        if not image:
+        if not image_str:
             raise InvalidImageName("Empty image repository/name")
-        if ("/" not in image) and using_default_registry:
-            image = default_repository + "/" + image
-        return image, tag
+        if ("/" not in image_str) and using_default_repository:
+            image_str = default_repository + "/" + image_str
+        return image_str, tag
+
+    @classmethod
+    def parse_image_str(cls, image_str: str, registry: str | None = None) -> ParsedImageStr:
```

Comment:
Is there any counterpart of this function?
like, parse `ImageRef` from `ParsedImageStr`.

Reply:
No, in my opinion, it wouldn’t be necessary since we can directly create an `ImageRef` object from an image string using `ImageRef.from_image_str`.